v23/requestid/model.go

```diff
@@ -0,0 +1,21 @@
+package requestid
+
+import (
+	"github.com/google/uuid"
+	"v.io/v23/context"
+)
+
+type contextKey int
+
+const requestIDKey = contextKey(iota)
+
+func WithNewRequestID(ctx *context.T) *context.T {
+	requestID := uuid.New()
+	return  context.WithValue(ctx, requestIDKey, requestID)
```

Comment:
with #154, you can just use return context.WithLoggingPrefix(ctx, requestID).

Reply:
Should we still include this so we support different loggers? 

For example, we may want to use a logger that logs in json format.

Reply:
I think we can both:
```
func WithNewRequestIDLoggingPrefix(ctx *context.T) *context.T {
	requestID, err := uuid.NewUUID()
	if err != nil {
		panic(err)
	}
	return  context.WithLoggingPrefix(ctx, requestID)
}
func WithNewRequestID(ctx *context.T) *context.T {
	requestID, err := uuid.NewUUID()
	if err != nil {
		panic(err)
	}
	return  context.WithValue(ctx, requestIDKey, requestID)
}
```

Reply:
yes, can certainly have both if you think there's a need for both, but the way you have the API setup now there would always be two different request IDs, so I'm curious to know what your use case for that it is? If you want to keep the WithNewRequestID then I'd suggest not having WithNewRequestIDLoggingPrefix and instead do something like:


```go
func WithNewRequestID(ctx *context.T) (*context.T, ID) {
	requestID, err := uuid.NewUUID()
	if err != nil {
		panic(err)
	}
	return  context.WithValue(ctx, requestIDKey, requestID), requestID
}

ctx, id = requestid.WithNewRequestID(ctx)
ctx = context.WithLoggingPrefix(ctx, id)
```

Or if you don't want to return the ID from `WithNewRequestID`
```go
ctx = requestid.WithNewRequestID(ctx)
ctx = context.WithLoggingPrefix(ctx, requestid.RequestID(ctx)) // get the existing request ID from the context.
```

Reply:
The use case I was considering is for a custom logger (e.g. json formatted logging). But, I now see that we just prepend the `requestid` to the list of `args` in `prefixedArgs()`, which should allow the logger to handle the args however it wants (including formatting). 

In that case, we shouldn't need `WithNewRequestID()` as you suggested.

Reply:
yes, that's correct. The custom logger would be implemented by replacing x/ref/internal/logger with a different package that implements your custom logger.